### PR TITLE
drop sysproxy.exe in net-core version

### PR DESCRIPTION
**Pull request by studentmain**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio
- [ ] Require translation update
- [ ] Require document update (readme.md, wikipage, etc)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New feature

---

### Description of your *pull request* and other information

An experimental p/invoke based system proxy implementation. Main goal is remove all blob in net-core version.

### CODE REVIEW

**Comment by chenshaoju** [comment]
This version looks need .NET Core to running.
image

But somehow, I installed `.NET Core Runtime 3.1.3`  ( https://dotnet.microsoft.com/download/dotnet-core/3.1 ), But it still unable running. :(

**Comment by studentmain** [comment]
> This version looks need .NET Core to running.
> image
> 
> But somehow, I installed `.NET Core Runtime 3.1.3` ( https://dotnet.microsoft.com/download/dotnet-core/3.1 ), It still unable running. :(

CI not works correctly, you need build it by yourself for now...

**Comment by chenshaoju** [comment]
My VS is broken, maybe someday...😅

File: shadowsocks-csharp/Util/SystemProxy/RAS.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    class RAS
+    {
+        public const int MaxEntryName = 256;
+        public const int MaxPath = 260;
+
+        const int ESuccess = 0;
+        const int RasBase = 600;
+        const int EBufferTooSmall = 603;
+
+        [DllImport("rasapi32.dll", CharSet = CharSet.Auto)]
```
Comment (celeron533) [inline shadowsocks-csharp/Util/SystemProxy/RAS.cs:38]:
```
// rasapi32.dll is the Remote Access API (RAS), used by Windows applications to control modem connections.
```

Comment (celeron533) [reply shadowsocks-csharp/Util/SystemProxy/RAS.cs:38]: rasapi32.dll exists since Windows 98. It is safe for win7 users

Comment (studentmain) [reply shadowsocks-csharp/Util/SystemProxy/RAS.cs:38]: As I mentioned in commit message, these code are mostly from shadowsocksrr.

Comment (studentmain) [reply shadowsocks-csharp/Util/SystemProxy/RAS.cs:38]: so we can expect same experience with them.

**Comment by studentmain** [comment]
going to rename net-core to v5/master @celeron533 @Stzx

**Comment by Stzx** [comment]
Okay, this should not cause any problems.
________________________________
发件人: Student Main <notifications@github.com>
发送时间: 2020年5月5日 11:16
收件人: shadowsocks/shadowsocks-windows <shadowsocks-windows@noreply.github.com>
抄送: Silence Tai <silence.m@hotmail.com>; Mention <mention@noreply.github.com>
主题: Re: [shadowsocks/shadowsocks-windows] drop sysproxy.exe in net-core version (#2874)


going to rename net-core to v5/master @celeron533<https://github.com/celeron533> @Stzx<https://github.com/Stzx>

―

**Comment by studentmain** [comment]
cc @database64128 @maikebing

**Pull request merged**